modules/common/condition/funcs_test.go

```diff
@@ -165,6 +165,55 @@ func TestSet(t *testing.T) {
 	g.Expect(c2.LastTransitionTime).To(BeIdenticalTo(time1))
 }
 
+func TestRemove(t *testing.T) {
+	conditions := Conditions{}
+
+	tests := []struct {
+		name      string
+		condition *Condition
+		want      Conditions
+	}{
+		{
+			name:      "Add nil condition",
+			condition: nil,
+			want:      CreateList(unknownReady),
+		},
+		{
+			name:      "Add unknownB condition",
+			condition: unknownB,
+			want:      CreateList(unknownReady, unknownB),
+		},
+		{
+			name:      "Add another condition unknownA, gets sorted",
+			condition: unknownA,
+			want:      CreateList(unknownReady, unknownA, unknownB),
+		},
+		{
+			name:      "Add same condition unknownA, won't duplicate",
+			condition: unknownA,
+			want:      CreateList(unknownReady, unknownA, unknownB),
+		},
+	}
+
+	g := NewWithT(t)
+
+	conditions.Init(nil)
+	g.Expect(conditions).To(haveSameConditionsOf(CreateList(unknownReady)))
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+
+			conditions.Set(tt.condition)
+
+			g.Expect(conditions).To(haveSameConditionsOf(tt.want))
+		})
+	}
+
+	conditions.Remove(unknownA.Type)
+	g.Expect(conditions.Has(unknownA.Type)).To(BeFalse())
+	g.Expect(conditions.Has(unknownB.Type)).To(BeTrue())
```

Comment:
OK, this is the meat of the test. But I suggest to have another case where you try to remove a type that is not in the lest.

Also try to remove the unknowReady as that is somewhat special.